compat/lazy-load-curl.c

```diff
@@ -0,0 +1,337 @@
+#include "../git-compat-util.h"
+#include "../git-curl-compat.h"
+#include <dlfcn.h>
+
+typedef void (*func_t)(void);
+
+#ifdef __APPLE__
+#define LIBCURL_FILE_NAME(base) base ".4.dylib"
+#else
+#define LIBCURL_FILE_NAME(base) base ".so.4"
+#endif
```

Comment:
Hmm. I just had a look and git-for-windows/git-sdk-32@d85efcecef95b40321c754ee3ed1dcd39838541d suggests that even for versions as old as v7.49, the number was a `4`...

Reply:
Actually, I don't think we're in _any_ kind of danger here. The `4` is the [libcurl ABI version](https://curl.se/libcurl/abi.html) which has not changed since v7.16.0, i.e. October 2006...

Reply:
It's not that simple. It shouldn't be a problem on Windows, but we might (unlikely but possible) encounter the 5 on other platforms.

See

https://github.com/curl/curl/commit/3f3d6ebe665f32f3fe6c3c49df86bdffd223090c

https://github.com/curl/curl/pull/1861

It's unlikely that we'll encounter this in the wild and we could address it when/if it happens.

Reply:
True. I added a code comment to help with this, should it become a problem.